common-libs/common/src/utils/objectUtil.ts

```diff
@@ -0,0 +1,67 @@
+export const buildObjectSkeletonFromPaths = (paths: string[] = []): Record<string, any> => {
+  const pathsAsObjects = paths.map(buildPathAsObject)
+  return mergeDeep({}, ...pathsAsObjects)
+}
+
+export const buildPathAsObject = (path: string): Record<string, any> => {
+  if (!path.includes('.')) {
+    return { [path]: {} }
+  }
+
+  const splitPath = path.split('.')
+  return splitPath.reduceRight((res, field) => ({ [field]: res }), {})
+}
+
+export const validateObjectHasPaths = (
+  object: Record<string, any>,
+  paths: string[] = [],
+): { field: string; path: string }[] => {
+  const errors: { field: string; path: string }[] = []
+  paths.forEach((path) => {
+    const splitPath = path.split('.')
+    let hasError = false
+    splitPath.reduce((fragment, field) => {
+      if (hasError) return undefined
+
+      if (fragment[field] === undefined || fragment[field] === null) {
+        errors.push({ field, path })
+        hasError = true
+        return undefined
+      }
+
+      return fragment[field]
+    }, object)
+  })
+
+  return errors.length > 0 ? errors : undefined
+}
+
+export const injectFieldForAllParentRoots = (
+  target: Record<string, any>,
+  field: string,
+  value: unknown,
+): Record<string, any> => {
+  const keys = Object.keys(target)
+  if (keys.length > 0) {
+    target[field] = value
+    keys.forEach((key) => injectFieldForAllParentRoots(target[key], field, value))
+  }
+
+  return target
```

Comment:
It could be much easier to understand with lodash:
```
if (!target.keys.length) { // what is the purpose of this condition?
  return {}
}

return {
  mapValues((targetValue) => injectFieldForAllParentRoots(targetValue, field, value))
  [field]: value,
}
```

Bonus point: `target` will no longer be mutated.

Other methods here can also be rewritten in a similar way.